src/persist/src/indexed/encoding.rs

```diff
@@ -92,6 +92,14 @@ pub struct BlobMeta {
     ///
     /// Invariant: Each stream id is in here at most once.
     pub traces: Vec<TraceMeta>,
+    /// Unsealed blobs to delete.
+    ///
+    /// Invariant: Each key is in here or pending_trace_deletes at most once.
```

Comment:
candidate invariant: none of the keys in this list are present in any `Unsealed`, and same for delete batches below. This should be pretty easily doable with a set intersection against the set of all unsealed/trace batch keys, which we compute in `validate` to check for global uniqueness?